### DISCUSSION TITLE
shutdown on k3s 1.24.4 takes too long 1min30s (systemctl timeout) even on KillMode=mixed

### DISCUSSION THREAD

**Discussion by hlacikd** [k3s-io/k3s#6045] [Q&A] [answered]
in releases prior to 1.23.7 it was enough to add KillMode=mixed to /etc/systemd/system/k3s.service , and when system shutdown executed, k3s killed containers and computer was turned off imediately 

For some reason unknown to me since 1.23.8 ---> up to current one 1.24.4 when doing so, it takes 90s again to shutdown system with k3s (... which is default systemctl timeout TimeoutStopUSec=1min 30s ... ), so KillMode mixed is ignored and k3s waits until timeout has passed to kill them ....

what has changed?

```
[Service]
Type=notify
EnvironmentFile=-/etc/default/%N
EnvironmentFile=-/etc/sysconfig/%N
EnvironmentFile=-/etc/systemd/system/k3s.service.env
KillMode=mixed
```

**Comment by brandond** [comment, marked as answer]
Duplicate of https://github.com/k3s-io/k3s/issues/2400 - see the discussion there.
